### PR TITLE
265: Adds cspell and removes markdown-spellcheck.

### PR DESCRIPTION
**Pull request by ianhundere**
as per: [Research alternative spell check packages#265](https://github.com/liatrio/devops-bootcamp/issues/265)

This PR also includes the necessary glob pattern for the markdown lint job. An additional ticket has been created to resolve those violations as well as reenable the glob pattern; `**/*.md`:
- [Resolve text-linter job violations#267](https://github.com/liatrio/devops-bootcamp/issues/267)

### CODE REVIEW

File: .cspell.json [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        "companies",
+        "css",
+        "docker",
+        "html",
+        "k8s",
+        "markdown",
+        "misc",
+        "npm",
+        "softwareTerms",
+        "useCompounds"
```
Comment (jburns24) [inline .cspell.json:112]: This feature alone is great

File: .cspell.json [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        "fontawesome/**",
+        "img/**",
+        "node_modules/**"
+    ],
+    "features": {
+        "weighted-suggestions": true
+    },
+    "allowCompoundWords": true,
+    "useGitignore": true
+}
```
Comment (jburns24) [inline .cspell.json:133]: NP: no new line. This makes github 😠

File: .cspell.json [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        "terraformrc",
+        "testvm",
+        "tfsdk",
+        "tfvars",
+        "theia",
+        "uncontainerized",
+        "uncordon",
+        "unstaged",
+        "westus",
+        "zxrnr"
```
Comment (jburns24) [inline .cspell.json:99]: I assume these are words we ignore. And I trust your judgement that they should be. But at a glance I am thinking "what?! how are `zxrnr` or `nqzxz` words"

Comment (ianhundere) [reply .cspell.json:99]: correct, this tells `cspell` that these words defined here are all good.

File: .cspell.json (hunk 2 above)
Comment (ianhundere) [reply .cspell.json:133]: usually default to what my linter does and my json linter removes any new lines ¯\_(ツ)_/¯

File: .cspell.json (hunk 3 above)
Comment (ianhundere) [reply .cspell.json:99]: more info:
`words` - a list of words to be considered correct.
`ignoreWords` - a list of words to be ignored (even if they are in the flagWords).

**Comment by jburns24** [comment]
@ianhundere If I am understanding your comment correct regarding the git hook. You added a pre-commit hook via the cspell docs. But that is only for your local repo correct? 

I think we can leverage something like [`husky`](https://typicode.github.io/husky/#/?id=automatic-recommended) to force this pre-commit hook for anyone with the repo. 

I would prefer this since most users will probably not install the dev dependencies or set up this pre-commit hook.

**Comment by ianhundere** [comment]
> I think we can leverage something like [`husky`](https://typicode.github.io/husky/#/?id=automatic-recommended) to force this pre-commit hook for anyone with the repo.

word, just included / tested it.

File: .github/workflows/text-linter.yml [hunk 4]
```diff
@@ -4,14 +4,16 @@ on:
...
     runs-on: ubuntu-latest
+    if: github.event.pull_request.draft == false
     steps:
       - uses: actions/checkout@v3
       - uses: DavidAnson/markdownlint-cli2-action@v9
         with:
           globs: |
             *.md
-            !test/*.md
+            !**/*.md
```
[the remark this answers is not in the document]
Comment (jburns24) [reply .github/workflows/text-linter.yml:19]: I see your comment in slack now. This is a chicken and an egg problem?

**Review by jburns24 — APPROVED** [review]
I see a comment in slack that the exclusion of all .md files in this pass was planned. In the future could you re-communicate this in the PR.

File: .github/workflows/text-linter.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (ianhundere) [reply .github/workflows/text-linter.yml:19]: just for posterity, below is the ticket that will focus on resolving the md violations:
- [Resolve text-linter job violations#267](https://github.com/liatrio/devops-bootcamp/issues/267)

i'll include a note in there in how to enable the glob again.

**Pull request merged**